### ISSUE TITLE
3th detected, but mouse can't move to 3th monitor

### ISSUE THREAD

**Issue by eriklely** [mgth/LittleBigMouse#80]
Partly the same as previous posters on 3rd monitor

In principle it works great, exactly what I was looking for. Only after I enable it I can't move my mouse to the 3rd monitor. It detects the 3rd monitor and I can configure it, but no configuration let's me move the mouse into this 3rd monitor.

Setup: 
left 1080 27 inch, middle 4k 32 inch, right 1080 27 inch
all samsung, left and right are equal

**Comment by PerplexedTrex** [comment]
I also had this issue.  When I load up the program, it recognizes 1 of the 2 samsung monitors and my Dell 144hz.  If I right click and place from windows config, it displays the other samsung but when I apply, it acts as if there were only 2.  I thought it might have to do with drivers since the 2 samsung were generic but after loading the driver one one and then the other, both resulted in the same outcome.

**Comment by servaram** [comment]
![image](https://user-images.githubusercontent.com/26465658/50010693-1d7a7580-ff88-11e8-9eee-b41c82736a97.png)
This is my current display setup, a gaming laptop hooked with 3 external monitors 1 4k directly via DisplayPort and 2 1080p monitors connected via USB 3.0, 

I am using "Little big mouse" works great, but had to move the overlapped monitors everytime I startup the program 
I had this program to autorun on windows boot, I always get the two 1080p monitors overlapped on each other, I have to move them manually every time, is this something can be fixed easily 

This is at windows startup 
![startup](https://user-images.githubusercontent.com/26465658/50010980-d5a81e00-ff88-11e8-8710-46196a762444.PNG)

I would have to move it every time like this 
![need_like this](https://user-images.githubusercontent.com/26465658/50011016-f1132900-ff88-11e8-9c28-680333c55b8a.PNG)

**Comment by trydalch** [comment]
I'm also experiencing this issue.

**Comment by davidAlittle** [comment]
I have this issue as well, but I can never get my mouse into the "hidden" screen. It shows up in the layout, but my mouse acts like it's not there.

**Comment by isherwood** [comment]
I have flanking monitors attached through a dock to my ThinkPad T570. The left one isn't accessible with LBM running. It does a beautiful job of overcoming the virtual wall on the right monitor, though.

**Comment by rewinded** [comment]
same issue here

**Comment by Solacegds** [comment]
Same issue here but I believe I have found the issue. The monitors that can't be found are all the same model as an existing monitor. I think you might be using a model name to identify them. If you switch to a driver key this might resolve it. Brilliant program tho.

**Comment by EricGrange** [comment]
I confirm Solacegds diagnostic: LBM worked great with my 3 monitors setup, up until my right monitor died, the replacement is the same model as the left one, and now LBM can no longer configure the original left monitor, and when reopening LBM I only see two monitors (middle and right).

I tried tweaking the description in the registry to make it different (the description is what LBM displays), but that was not enough.

**Comment by ItsANoBrainer** [comment]
Also confirming Solacedgs' information. I have a 3 monitor setup with my two side monitors being the exact same. It seems to stack these two on top of each other on program restart/startup thinking they are the same one. Hopefully you can fix this so I can use this amazing program.

**Comment by FumbarHuygens** [comment]
Confirming that this is still an issue. This software is invaluable and is unfortunately rendered unusable by this bug.

**Comment by Tom-van-Grinsven** [comment]
Yes still an issue. I agree with FumbarHuygens about this bug. It would solve exactly the problem I have with my multi monitor setup if this bug is fixed.

**Comment by Priva-au** [comment]
This is still an on-going issue, is this likely to be fixed or has all updates ceased for LBM? 
If it's unlikely to be fixed does anyone have any alternative solutions?

**Comment by Sputnikcer** [comment]
I've posted a hotfix for this issue on https://github.com/mgth/LittleBigMouse/issues/132

**Comment by IvanZdravkovBG** [comment]
I have the same issue. A Horizontal large monitor and 2 identical vertical small ones. Mouse can't enter the third monitor.

**Comment by gribuser** [comment]
Same thing with me. GUI works just fine, when in comes to real mouse work all is messed up :(

**Comment by gribuser** [comment]
Actually, it seemse like the App had stored some settings at the first run, and then uses them whatever you change in GUI. So I guess internally all is fine, it's just messed up with the wrong setup.

**Comment by gribuser** [comment]
Even worse, the problem is that I have two identical monotors, so, as I get it, it's writihg theirs config to the same key in the registry. So while having 3 monigors in GUI I only get two keys in the config (one of the keys is overwritten). That's how things are messed up (looks quite easy to fix as app knows all monitor numbers of UUID's or whatever).

**Comment by polykoma** [comment]
Dear creator. Thank you for the awesom tool. Unfortunately after getting a third screen. Now having 1x main  and 2x identical sides the tool is unusable since the issue which got elaborated in the previous post.
We would love to see a fix since its quite easy to implement.
Thank you for your time!

**Comment by JonathanWillis** [comment]
Hi,

Just to tag in, i also have the same issue, but i didn't have it initially. I added to new 4k monitors to my setup, but my display port cable wasn't long enough, so i used a HDMI cable instead. In this instance i had no problem. My longer display port cable has arrived, and now i get the same issue as others above.

So a potential workaround for some, could be to use a HDMI cable for one monitor and a display port cable for the other identical monitor, then it works fine.

**Comment by polykoma** [comment]
> 
> 
> Hi,
> 
> Just to tag in, i also have the same issue, but i didn't have it initially. I added to new 4k monitors to my setup, but my display port cable wasn't long enough, so i used a HDMI cable instead. In this instance i had no problem. My longer display port cable has arrived, and now i get the same issue as others above.
> 
> So a potential workaround for some, could be to use a HDMI cable for one monitor and a display port cable for the other identical monitor, then it works fine.

Im not sure if I can confirm.
Unfortunately my both second/third screens are identical and have only HDMI input.

I connected as follow:  1x DP  to DP Main Screen  1x HDMI to HDMI for second screen 1x DP to HDMI on third screen.
Maybe it would work if my seconds had DP input aswell. Interesting.

**Comment by novxa** [comment]
Hi, 

I'm facing the same issue (cant move mouse to identical monitor) and fixed it on my end by changing the HardwareID key in the registry for one of my monitors. Software works flawlessly now but I'm not sure if this will give me other issues down the road, would still love to see a fix

**Comment by JonathanWillis** [comment]
The project is on github and looks to be c# I'll have to take a look when I
get chance but as I'm starting a new job in react/aws and learning on the
job, digging into this will take a back seat.

It sounds simple enough, somewhere you should be able to assign a unique
I'd per monitor but how that could ripple though the application I do not
know.

Honestly windows should not develop paint3d and instead improve these exact
things as multi monitor is so common, for myself at home and work I've had
2 monitors since the days of windows xp.

I also wish they would render the application once per DPI, so when you
drag a monitor between windows the application doesn't grow or shrink. And
whilst they are at it, even with the latest improvements, I still find many
applications even task manager have blurry text on 4k screens, although
overriding DPI to system (enhanced) can help. Rant over :)

**Comment by polykoma** [comment]
> 
> 
> Hi,
> 
> I'm facing the same issue (cant move mouse to identical monitor) and fixed it on my end by changing the HardwareID key in the registry for one of my monitors. Software works flawlessly now but I'm not sure if this will give me other issues down the road, would still love to see a fix

Hey - can you shortly advise which tool you used to change the HWID? Would love to try this fix!

**Comment by JonathanWillis** [comment]
Can't be fixed without the creator of the project, i pulled the code down, but it won't build as not all the code is on Github.  mgth (the creator) says they hope to make the code available later this year.

**Comment by novxa** [comment]
> > 
> > 
> > Hi,
> > 
> > I'm facing the same issue (cant move mouse to identical monitor) and fixed it on my end by changing the HardwareID key in the registry for one of my monitors. Software works flawlessly now but I'm not sure if this will give me other issues down the road, would still love to see a fix
> 
> Hey - can you shortly advise which tool you used to change the HWID? Would love to try this fix!

Hey, sorry only seeing this now. I changed it in regedit which is a built in windows tool. It's not perfect because I have to do it each time I reboot

**Comment by ashencloud** [comment]
damn, I just got my third monitor and I excitedly plugged it in only to see that my 3rd monitor is inaccessible :(

**Comment by polykoma** [comment]
> 
> 
> damn, I just got my third monitor and I excitedly plugged it in only to see that my 3rd monitor is inaccessible :(

Welcome to the club. You can just deinstall and wait for new version....

**Comment by voidptr** [comment]
> > > Hi,
> > > I'm facing the same issue (cant move mouse to identical monitor) and fixed it on my end by changing the HardwareID key in the registry for one of my monitors. Software works flawlessly now but I'm not sure if this will give me other issues down the road, would still love to see a fix
> > 
> > 
> > Hey - can you shortly advise which tool you used to change the HWID? Would love to try this fix!
> 
> Hey, sorry only seeing this now. I changed it in regedit which is a built in windows tool. It's not perfect because I have to do it each time I reboot

Can you elaborate? Where in the registry did you find the key?

**Comment by voidptr** [comment]
Ok, I can confirm novxa's workaround. 

1. Start LBM, and note the name of the "duplicate" display. It appears in the top left corner of the first LBM screen. In my case, it was GSMB09. This is the duplicate key that's ultimately troublesome.
2. Open Regedit, and navigate to `Computer\HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Enum\DISPLAY\<your monitorname>`. You will probably see several folders underneath. Each is a monitor configuration with that same key. My subkeys have names like: `5&f3b8925&0&UID4609`
3. Pick one and click on it. Then, change the `HardwareID` value to something else. Mine was originally `MONITOR\GSM5B09` and I changed it to `MONITOR\GSM5B10`
4. Restart LBM. Your second monitor should now appear, ,named whatever you chose. Mine is now `GSM5B10`
5. Rejoice!

**Comment by polykoma** [comment]
> 
> 
> Ok, I can confirm @novxa's workaround.
> 
>     1. Start LBM, and note the name of the "duplicate" display. It appears in the top left corner of the the monitor, on the first LBM screen. In my case, it was `GSMB09`. This is the duplicate key that's ultimately troublesome.
> 
>     2. Open Regedit, and navigate to `Computer\HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Enum\DISPLAY\<your monitorname>`. You will probably see several folders underneath. Each is a monitor configuration with that same key. My subkeys have names like: `5&f3b8925&0&UID4609`
> 
>     3. Pick one and click on it. Then, change the `HardwareID` value to something else. Mine was originally `MONITOR\GSM5B09` and I changed it to `MONITOR\GSM5B10`
> 
>     4. Restart LBM. Your second monitor should now appear, named whatever you chose. Mine is now `GSM5B10`
> 
>     5. Rejoice!
> 
> 
> I can now configure everything how I had it, with all the weird workarounds (gaps between all displays, enabling `Allow Discontinuity`), and everything works! Hooray! Thank you @novxa for the breadcrumb!

Does it stay like that after reboot ?

**Comment by gribuser** [comment]
> Does it stay like that after reboot ?
Yes, it works just fine after reboot.
I fear things will fall apart after some monitors on/off/sleep, i.e. when hardware is renumbered. My PC seem to beleive I unplug and plug minitors daily. This remains to be seen, but after normal reboot everything is ok.

**Comment by polykoma** [comment]
> 
> 
> > Does it stay like that after reboot ?
> 
> Yes, it works just fine after reboot.
> I fear things will fall apart after some monitors on/off/sleep, i.e. when hardware is renumbered. My PC seem to beleive I unplug and plug minitors daily. This remains to be seen, but after normal reboot everything is ok.
> 
> ps. Hack seem to endure anything, so I am quite happy now/

Just did this and works like a charm (for now). Thank you for the early christmas Stranger! :> <3

**Comment by Nexab34** [comment]
> Ok, I can confirm @novxa's workaround.
> 
> 1. Start LBM, and note the name of the "duplicate" display. It appears in the top left corner of the the monitor, on the first LBM screen. In my case, it was `GSMB09`. This is the duplicate key that's ultimately troublesome.
> 2. Open Regedit, and navigate to `Computer\HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Enum\DISPLAY\<your monitorname>`. You will probably see several folders underneath. Each is a monitor configuration with that same key. My subkeys have names like: `5&f3b8925&0&UID4609`
> 3. Pick one and click on it. Then, change the `HardwareID` value to something else. Mine was originally `MONITOR\GSM5B09` and I changed it to `MONITOR\GSM5B10`
> 4. Restart LBM. Your second monitor should now appear, named whatever you chose. Mine is now `GSM5B10`
> 5. Rejoice!
> 
> I can now configure everything how I had it, with all the weird workarounds (gaps between all displays, enabling `Allow Discontinuity`), and everything works! Hooray! Thank you @novxa for the breadcrumb!

Tried this. For me, only editing the 3rd (had 4) subkey HardwareID actually changed the number on the top left. But even though all the monitors automatically show up now, I still can't scroll over to my 3rd monitor.
https://imgur.com/j4irzlR

**Comment by polykoma** [comment]
I need to tell you guys that after a reboot the hardware ID is reset to previous state on my win10 machine with latest updates rendering the workaround nearly useless for daily usage

**Comment by novxa** [comment]
I want to add another issue to the workaround. I've recently been having a bug with shadowplay (nvidia screen recording software) capturing my wrong display. After finding [this reddit post](https://www.reddit.com/r/GeForceExperience/comments/imes94/fix_recording_the_wrong_monitor_on_shadowplay/ ), I'm quite certain it's because of messing with the registry. 

Hope we can find another solution but for now I have to stop using this I think

**Comment by polykoma** [comment]
Any solution meanwhile? @mgth

**Comment by polykoma** [comment]
> 
> 
> I did find a solution to this but it involves hardware and money (about £20). I ended up buying a 4K HDMI Pass-Through EDID Emulator from Amazon. This presents a different brand of monitor to the machine and therefore LBM and sorts out the issue assuming you only have one duplicated monitor brand.

NICE ! Bought such said Emulator from LogiLink on Amazon.de for 8 bucks and it works! Sweet! Thank you for sharing!

**Comment by eco9898** [comment]
Been having the same issue, can confirm two monitors of the same type will be treated as one and placed on top of each other, causing one to become unreachable

**Comment by eco9898** [comment]
identifying monitors via a combination of their serial number and model number should stop this issue

**Comment by JosiahWhite** [comment]
Another solution to this problem is to patch the `_idMonitor` getter from the `Monitor` class in `HLab.Windows.Monitors.dll`
This getter has specific cases to deal with a serial number that is "1234567890123" and instead use another item as a unique identifier, but no case for the serial number being empty.

Using dnSpy, you can duplicate this check and add checks for an empty serial number or a serial number not existing. I was able to do this and the issue is solved permanently. No longer do I need to add a .reg file on startup.

This is what the IL code looks like after duplicating the existing check down twice: 
![image](https://user-images.githubusercontent.com/834186/146206176-12672633-d62d-4de7-a8da-b5a85249aa80.png)

**Comment by FumbarHuygens** [comment]
> Another solution to this problem is to patch the `_idMonitor` getter from the `Monitor` class in `HLab.Windows.Monitors.dll` This getter has specific cases to deal with a serial number that is "1234567890123" and instead use another item as a unique identifier, but no case for the serial number being empty.
> 
> Using dnSpy, you can duplicate this check and add checks for an empty serial number or a serial number not existing. I was able to do this and the issue is solved permanently. No longer do I need to add a .reg file on startup.
> 
> This is what the IL code looks like after duplicating the existing check down twice: ![image](https://user-images.githubusercontent.com/834186/146206176-12672633-d62d-4de7-a8da-b5a85249aa80.png)

This is super helpful. Could you elaborate a bit on where exactly you are finding these lines? I was able to navigate to the `Monitor` class but the `get_idMonitor` section under `IdMonitor` is only like four lines executing `this._idMonitor.get()`.

The `_idMonitor` section at the very bottom is a little longer but still doesn't look like what you posted.

Thanks!


![image](https://user-images.githubusercontent.com/39710491/156469031-93c3c918-63ec-448a-9ec7-5483c880052d.png)

![image](https://user-images.githubusercontent.com/39710491/156469433-b7ec2875-f743-4d2b-a173-9f6fa56c4fee.png)

**Comment by JosiahWhite** [comment]
So I had to modify the code in IL mode, to do this I selected IL from the top dropdown, and then I scrolled down to roughly line 315 in the source code to locate the getter in the IL code. 
![image](https://user-images.githubusercontent.com/834186/156486604-fb6ebb95-6f46-4f3f-a5bc-26e6429eeeb2.png)


After this, you can right click and hit "Edit IL Instructions" anywhere inside of that method. 
![image](https://user-images.githubusercontent.com/834186/156486468-43cc9372-057c-464a-a64e-53cd075d9c79.png)


I have uploaded my modified DLL if anyone would like to use it. Feel free to analyze, but all the changes I made are documented above :)
To install, simply overwrite the existing DLL in "C:\Program Files\LittleBigMouse" and restart your application.
[HLab.Windows.Monitors.zip](https://github.com/mgth/LittleBigMouse/files/8174552/HLab.Windows.Monitors.zip)

**Comment by ItsANoBrainer** [comment]
Can confirm @JosiahWhite 's DLL patches this issue. You're a legend, thank you!

VirusTotal for anyone who is skeptical: 
https://www.virustotal.com/gui/file/5da18a639b0713a81fb2baf55af3844b3ca43c49fda5bc43ee40484f0d2e7cbb?nocache=1

**Comment by Stari89** [comment]
I can also confirm that @JosiahWhite works! Even after restart. Cheers!

**Comment by ifexception** [comment]
I was experiencing the same issue discussed here. For me, it was only picking up my two monitors, but confusingly, one of them was being "mapped" to the laptop screen so when I applied settings, the mouse would only move between the laptop screen and one monitor.

By chance, I clicked on "Info" and then you get the monitors with all their settings and names. Right clicking on it brings up this menu: 
![image](https://user-images.githubusercontent.com/17222073/202475475-81384f75-2def-4aa3-b9c6-7c83058620fd.png)

Clicking "Place from window config" seemed to have refreshed the settings and all three monitors got picked up and the mouse moves through all three monitors with _no_ issues

**Comment by Arnedeklerk** [comment]
@Samconboy You're right, I can't get this to work on W11 either. Any luck?

**Comment by Arnedeklerk** [comment]
@FumbarHuygens 

Any idea why this isn't working on Windows 11? :)

Was so happy to find this tool, just for it to not work the day after I update Windows 🗡️

**Comment by DailyClockwork** [comment]
> Can't get it to work with the replaced DLL, the control.exe silently crashes.
> 
> For reference am running win11 with LBM version: LittleBigMouse_4.2.7124.42685
> 
> Debugging with DnSpy shows its trying to reach out to a network?? Doesn't seem right?
> 
> ![image](https://user-images.githubusercontent.com/35070146/160682100-ac66619c-6bbe-4706-ac5a-1251db2f7aca.png)
> 
> Am going to attempt to modify the dll myself - need to get it to not use hardwareId for my monitors but no idea what I'm doing currently...

did you ever figure out how to do it yourself?

**Comment by Arnedeklerk** [comment]
@DailyClockwork 

I wound up back on win 10!

But have found that during some tasks (such as gaming) that frequent double clicks or holding down of mouse 1 happens when littlebigmouse is active. So, not ideal either.

**Comment by Samconboy** [comment]
I gave up on trying to get it working, I'm not experienced enough to know what I'm looking for with this sort of thing and my layout worked well enough I got used to not using LBM

**Comment by strazto** [comment]
Just confirming that https://github.com/mgth/LittleBigMouse/issues/80#issuecomment-1057613055
> I have uploaded my modified DLL if anyone would like to use it. Feel free to analyze, but all the changes I made are documented above :) To install, simply overwrite the existing DLL in "C:\Program Files\LittleBigMouse" and restart your application. [HLab.Windows.Monitors.zip](https://github.com/mgth/LittleBigMouse/files/8174552/HLab.Windows.Monitors.zip)

Fixed the problem for me.

Looks like @mgth has been active on this project as of recent months, hopefully that means they'll be adding the necessary components that allow other users to build this project themselves so community contributions can be feasible

**Comment by Samconboy** [comment]
@strazto I did try that one but the program would silently crash hencgme my post about it. Hopefully if mgth is working on it they can fix! TIA!!

**Comment by M-SS-NGNUMB-R** [comment]
Anyone ever get this figured out? Just bought another monitor of the same model as a previous one, and the updated DLL just crashes the program.

**Comment by M-SS-NGNUMB-R** [comment]
I'm running into this same issue after buying a second of the same model monitor, and the DLL just crashes the program for me. I admittedly have no idea what I'm doing when it comes to dnspy; I can get the registry editing solution working, but I'd really like to get the DLL solution working so I don't have to re-edit those keys every time I restart the PC.
I tried skimming through the two DLLs side by side to see if I could pinpoint the differences between them, but as far as I can tell, they're basically the same.
Is it possible to re-apply these same changes to a clean DLL and get the same effect without the crash? And if so, would anyone be able to point me in the right direction? After comparing the clean DLL, modified DLL, and the screenshots posted here, I'm not sure what checks are being duplicated, or where.
I'm on Windows 10, using LittleBigMouse version 4.2.7124.42685.

**Comment by IvonTSaldivar** [comment]
@mgth I saw you marked this as completed - has this fix made it into a release? The issue still occurs for me on Windows 11, on v5.0-beta1. I have two of the same monitor plugged in, and while LBM is running, I can't move my mouse onto one of them even though it shows up in the LBM settings..

**Comment by strazto** [comment]
> has this fix made it into a release yet? The issue still occurs for me on Windows 11, on v5.0-beta1.

The v5 beta release corresponds to https://github.com/mgth/LittleBigMouse/commit/395a6304e14361c838833c4e130ea23d43f013cb

Based on the commit history, that's a very fresh build, and nothing after that commit appears to pertain to your issue 

It's possible that @mgth fixed a different issue that sounds related from the issue title, since most people here are reporting issues with running two of the same device model

**Comment by polykoma** [comment]
I tried to install the new version but I get an error while installing on windows 10?!

And yes correct the issue here is when having 2x identical model tool doesnt work cuz it cant distinguish that you got 2 since it relies on model as identifier.

WE NEED A FIX FOR THIS PLEASE <3

**Comment by mgth** [comment]
Hello, very sorry for the loooong delay on this, everyone having this issue, may you post a config export there ? (by clicking on the export (gear) button at the right bottom). Thank you.

**Comment by ItsANoBrainer** [comment]
> Hello, very sorry for the loooong delay on this, everyone having this issue, may you post a config export there ? (by clicking on the export (gear) button at the right bottom). Thank you.

I am using the fix by JosiahWhite, so I don't know if my config is affected, but I have a 3 monitor setup with my two side monitors being the exact same. My main monitor is a AOC27" 1440p 144hz, and my side monitors are AOC24" 1080p 144hz.

https://pastebin.com/ic5jGtHy

**Comment by polykoma** [comment]
Thank you so much for reopening.
The culprint is that the tool relies on an Identifier of the Monitor (probably the Model Type ID) and when you have 2x the IDENTICAL model the software can not save settings because settings only make one entry or smth.

Thats what I assume but Im no dev.

**Comment by IvonTSaldivar** [comment]
Here's my config export from 5.0.3.0:
[LittleBigMouse.Export.gz](https://github.com/mgth/LittleBigMouse/files/13590745/LittleBigMouse.Export.gz)

**Comment by IvonTSaldivar** [comment]
Release 5.0.4.1 does fix the problem for me! Thank you!
